src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java

```diff
@@ -53,15 +67,46 @@ public FilesBrowserViewImpl(
 
   @Override
   public void configure(String entityId) {
+    boolean isReactFileBrowser = featureFlagConfig.isFeatureEnabled(
+      FeatureFlagKey.REACT_FILE_BROWSER
+    );
     title.setVisible(false);
-    entityTreeBrowser.configure(entityId);
+    reactFileBrowserContainer.setVisible(isReactFileBrowser);
+    files.setVisible(!isReactFileBrowser);
+    if (isReactFileBrowser) {
+      rerenderFileBrowser(entityId);
+    } else {
+      entityTreeBrowser.configure(entityId);
+    }
+  }
+
+  public void rerenderFileBrowser(String parentContainerId) {
+    // TODO: Should not need to remount, but currently the parentContainer is not shown if the existing react component is updated
+    // TODO: Bug: The tree is not always auto-expanding (to the path to the parent container Id)
+    reactFileBrowserContainer.clear();
+    ReactComponent componentContainer = new ReactComponent();
```

Comment:
You shouldn't need to create a new ReactComponent instance, you should just be able to create one instance and call `render` on it with the new ReactElement. But it's possible you tried that and it resulted in another bug.